runtime/compiler/env/VMJ9Server.cpp

```diff
@@ -1676,13 +1689,15 @@ TR_OpaqueClassBlock *
 TR_J9SharedCacheServerVM::getClassFromSignature(const char * sig, int32_t sigLength, TR_ResolvedMethod * method, bool isVettedForAOT)
    {
    TR_ResolvedRelocatableJ9JITaaSServerMethod* resolvedJITaaSMethod = (TR_ResolvedRelocatableJ9JITaaSServerMethod *)method;
-   return getClassFromSignature(sig, sigLength, (TR_OpaqueMethodBlock *)resolvedJITaaSMethod->getPersistentIdentifier(), isVettedForAOT);
+   J9ClassLoader * cl = resolvedJITaaSMethod->getClassLoader();
```

Comment:
Why is the change needed here. The newly created getCacheClassFromSignature() only use the 'per-compilation' cache.